config/routes.rb

```diff
@@ -1,7 +1,10 @@
 Rails.application.routes.draw do
 
-  root 'mkdev#index'
-
-  resources :cards
+  root "main#index"
+  resources :cards do
+    member do
+      post 'compare'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.